### PR TITLE
[Azure.Core] Update options to use Format string

### LINKED ISSUE #37466: [Azure.Core] Collapse IgnoreReadOnlyProperties and IgnoreAdditionalProperties into Format string

### PR DESCRIPTION
**Pull request by nisha-bhatia**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/37466
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/core/Azure.Core/src/Serialization/ModelSerializerOptions.cs [hunk 1]
```diff
@@ -35,5 +40,14 @@ public class ModelSerializerOptions
         /// NameHint for Xml Models
         /// </summary>
         public string? NameHint { get; set; }
+
+        private string ValidateFormat(string x)
+        {
+            if (x != "D" && x != "W")
```
Comment (m-nash) [inline sdk/core/Azure.Core/src/Serialization/ModelSerializerOptions.cs:46]: Can we make constants for these so we know what they mean?

Comment (christothes) [reply sdk/core/Azure.Core/src/Serialization/ModelSerializerOptions.cs:46]: If `Format` can't be used with the other standard string formats, should we consider doing this as an Enum instead?

File: sdk/core/Azure.Core/api/Azure.Core.net461.cs [hunk 2]
```diff
@@ -1145,12 +1144,17 @@ public static partial class ModelSerializer
...
-        public ModelSerializerOptions() { }
-        public bool IgnoreAdditionalProperties { get { throw null; } set { } }
-        public bool IgnoreReadOnlyProperties { get { throw null; } set { } }
+        public ModelSerializerOptions(string format = "Data") { }
+        public string FormatType { get { throw null; } }
         public string? NameHint { get { throw null; } set { } }
         public bool PrettyPrint { get { throw null; } set { } }
         public System.Collections.Generic.Dictionary<System.Type, Azure.Core.Serialization.ObjectSerializer> Serializers { get { throw null; } }
+        [System.FlagsAttribute]
+        public enum Format
```
Comment (m-nash) [inline sdk/core/Azure.Core/api/Azure.Core.net461.cs:1153]: This should be implemented as an [extensible enum](https://azure.github.io/azure-sdk/dotnet_implementation.html#dotnet-enums) with values of Wire = "W" and Data = "D"

File: sdk/core/Azure.Core/src/Serialization/ModelSerializerOptions.cs [hunk 3]
```diff
@@ -12,18 +12,32 @@ namespace Azure.Core.Serialization
...
+            //throw ArgumentException if not "D" or "W"
+            ModelSerializerFormatKind = ValidateFormat(format);
+        }
 
         /// <summary>
-        /// Bool that determines if AdditionalProperties will be serialized. Default is false.
+        /// String that determines Format of serialized model. "D" = data format which means IgnoreReadOnly and IgnoreAdditionalProperties are false, "W" = wire format which means both properties are true. Default is "D".
         /// </summary>
-        public bool IgnoreAdditionalProperties { get; set; }
+        public string ModelSerializerFormatKind { get; }
```
Comment (m-nash) [inline sdk/core/Azure.Core/src/Serialization/ModelSerializerOptions.cs:27]: This should be type ModelSerializerFormat

File: sdk/core/Azure.Core/tests/public/ModelSerializationTests/UsingJsonSerializerTests.cs [hunk 4]
```diff
@@ -15,8 +15,6 @@ public class UsingJsonSerializerTests
         public void SerializeTest(bool ignoreReadonlyProperties)
         {
             string expected = "{";
-            if (!ignoreReadonlyProperties)
```
[the remark this answers is not in the document]
Comment (nisha-bhatia) [reply sdk/core/Azure.Core/tests/public/ModelSerializationTests/UsingJsonSerializerTests.cs:18]: let me restore these changes

File: sdk/core/Azure.Core/src/Serialization/ModelSerializerOptions.cs [hunk 5]
```diff
@@ -12,18 +12,32 @@ namespace Azure.Core.Serialization
...
         /// <summary>
         /// Dictionary that holds all the serializers for the different model types.
         /// </summary>
         public Dictionary<Type, ObjectSerializer> Serializers { get; } = new Dictionary<Type, ObjectSerializer>();
+
+        private string ValidateFormat(string x)
+        {
+            if (x != ModelSerializerFormat.Data && x != ModelSerializerFormat.Wire)
+            {
+                throw new ArgumentException("Format must be either 'ModelSerializerFormat.Data' or 'ModelSerializerFormat.Wire'.");
```
Comment (m-nash) [inline sdk/core/Azure.Core/src/Serialization/ModelSerializerOptions.cs:38]: need to interpolate these enum values

**Pull request merged**